code/src/java/pcgen/rules/persistence/token/AbstractSimpleChooseToken.java

```diff
@@ -102,9 +102,10 @@ protected ParseResult parseTokenWithSeparator(LoadContext context,
 		}
 		else
 		{
-			if (hasIllegalSeparator('|', activeValue))
+			ParseResult pr = checkForIllegalSeparator('|', activeValue);
+			if (!pr.passed())
 			{
-				return ParseResult.INTERNAL_ERROR;
+				return pr;
```

Comment:
Pattern seems a little unusual that you only return if it hasn't passed.  Why wouldn't it also be returned if it did pass?

Reply:
So the ParseResult in this case is simply a check of "did this have legal separators?"  If it failed, then we need to let someone know why.  If it succeeded, it only meant the separators were legal - we still need to parse the token to get the content.

I'm open to a better structure here, but I'm also operating under the principle of "Don't throw exceptions to send back a normal condition failure"... 

I can't reasonably go to a design that checkForIllegalSeparator takes in a 3rd argument which it calls to "continue", as that would produce a wild set of temporary objects that would obfuscate what is really happening. 

I don't want to inline checkForIllegalSeparator because it's very standardized code, multiple if statements, and used hundreds of times, so it really does belong in a small method by itself.

I'm open to options - I'm not a huge fan of doing this, but I haven't thought of a better structure to use in Java.  What's the appropriate design pattern for that type of scenario?

Reply:
Std thing would be to throw an `IllegalSeparatorsException` as this is an 'exceptional case' (we expect LST data separators to be in the correct format).  Digging into `ParseResult`... Oh I see now.  We're collecting all parse errors to return to the user so we don't bomb out on the first one.  And we're using an internal QueueMessage attached to that interface.  OK, you'd typically want a CQRS / eventbus / message bus put in place here, the queue serves that purpose (a little crudely but OK).  I totally agree with having `checkForIllegalSeparator` extracted.

TLDR - lets leave this as is as a refactor to an event bus system would be global (out of scope for this).